### ISSUE TITLE
how to train the network

### ISSUE THREAD

**Issue by eugenelyj** [uzh-rpg/rpg_e2vid#1]
Thanks for sharing of this nice work!! But could I train this network by myself?

**Comment by supitalp** [comment]
Hi,
Unfortunately, for various reasons we have not released the training code or training data for this work.
However, this code provides the network model, so it should be relatively straightforward to implement the training procedure described in the paper on top of that.
In addition, the event camera simulator that we used to generate synthetic training data is available open source as well (https://github.com/uzh-rpg/rpg_esim), so it should be possible to generate your own training data as well.

**Comment by eugenelyj** [comment]
@supitalp Thank you. And i saw that in [this paper](https://arxiv.org/abs/1906.07165) you use optical flow to construct temporal consistency loss, right? As i know, esim can output optical flow in a fixed rate, but in [this network](https://github.com/uzh-rpg/rpg_e2vid/), input are fixed number of events so the input rate is not fixed, how do you deal with it?

**Comment by supitalp** [comment]
Hello,
Good point. Yes, that is correct, optical flow is used at train time to construct the temporal consistency loss.
The network can actually be run either with a fixed number of events (as described in the paper), or with a fixed duration window. We train the network with windows of fixed duration in simulation (with windows of 33 ms).
Note that we do not use directly the optic flow maps from ESIM. Indeed, those are instantaneous optic flow maps (i.e. they describe the instantaneous pixel velocity in pixels per second at each pixel), while what we need for training are "displacement maps" (i.e. that describe how to displace every pixel from one image to its position in the next image).
To get those, we use the depth maps + relative camera poses from ESIM (which we output at 30 Hz), and compute the displacement maps from these two quantities).

Hope that helps, please let me know if something is not clear.
